code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -109,8 +109,9 @@
 	var/consumed = 0
 	switch(air.temperature)
 		if(-INFINITY to WATER_VAPOR_DEPOSITION_POINT)
-			if(location?.freeze_turf())
-				consumed = MOLES_GAS_VISIBLE
+			location?.freeze_turf()
+			return REACTING
```

Comment:
this means cold watervapor will always be active, even if nothing is going on.
this is troublesome for not wasting processing time, especially since fire creates water currently. 

what is the motivation behind this change? is it just to make hydrogen production more consistent?

Reply:
We can't make the turf being frozen already return a non reaction without making turfs unfreezing/freezing trigger activity. maybe that's the best solution here?

Reply:
it is also to address a inconsistency with gas behavior vs information. the water condensation reaction explicitly states it does not consume moles below 200kelvin. false information makes me angry.

Reply:
> We can't make the turf being frozen already return a non reaction without making turfs unfreezing/freezing trigger activity. maybe that's the best solution here?

Turfs freezing will also freeze everything inside the turf. We could create a freezing component (or element) or something that listens to things entering the turf and freezes them, then removes itself when it gets too hot or there isn't enough water vapour. We can then just make the reaction not return REACTING if the turf already has the freezing component.

Also it looks like you're using the webeditor. I recommend setting up Visual Studio Code and git.

Reply:
I know how to run the server for testing, and managed to do so before undrafting the second time. i don’t quite know my way around the debug tools, so  the testing I was able to do was limited.

Reply:
My concern isn't lag it's just if it works. I need this changed regardless of lag though. We cannot have reactions that go on forever.

Reply:
if you got any suggestions on how to check for frozen turfs, i’m all ears. I’m not good at locating examples of procs outside of the files I have touched.

Reply:
it would be done using a trait on the turf. I think what pickle is suggesting is maybe a bit outside the scope of a first pr, but if you are interested and want help feel free to bug me on discord, I'm @thefinalpotato